python/pyarrow/parquet/core.py

```diff
@@ -16,23 +16,22 @@
 # under the License.
 
 
-from collections import defaultdict
-from concurrent import futures
-from contextlib import nullcontext
-from functools import partial, reduce
-
-import sys
-import json
-from collections.abc import Collection
-import numpy as np
-import os
-import re
-import operator
-import urllib.parse
-import warnings
-
-import pyarrow as pa
-import pyarrow.lib as lib
+from collections import defaultdict as _defaultdict
```

Comment:
Is there a reason to rename those? I thought that if `__all__` is specified as now done in this PR, that the `import *` in `__init__.py` would only import the elements in `__all__` regardless of other names being present that don't start with an underscore?

Reply:
I was mirroring what was done in pyarrow.__init__.py, with the thought that once `_filters_to_expression` is either deprecated to use `filters_to_expression` or subsequently removed, then we can remove `__all__` and follow the same design as `pyarrow.__init__`.

Reply:
but I do agree, renaming isn't needed if we are just going to keep `__all__` defined in this module.

Reply:
Given that this is for importing all symbols in a submodule (core.py) to the main public module (pyarrow.parquet), I think the `__all__` approach is fine, and we can keep doing that. 
(so I would revert the other renamings)

Reply:
Removed that commit, should we keep the `_filters_to_expression` renaming deprecation?

Reply:
I am fine with deprecating it, but I think we should initially do that with a DeprecationWarning instead of FutureWarning (those are less visible, and since this is a function that is used by other packages, not by end-users, ideally first those packages can solve the warning without their users seeing it).

Reply:
Good point. I updated the `_deprecate_api` to accept the type to pass to `.warn()`. Think this should be good now.